services/sidecar/src/simcore_service_sidecar/cli.py

```diff
@@ -30,8 +32,20 @@ def main(
         log.exception("Uncaught exception")
 
 
+async def perdiodicaly_check_if_aborted(is_aborted_cb: Callable[[], bool]) -> None:
+    log.info("Starting periodic check of task abortion...")
+    while await asyncio.sleep(SIDECAR_INTERVAL_TO_CHECK_TASK_ABORTED_S, result=True):
+        if is_aborted_cb():
+            log.info("Task was aborted. Cancelling...")
+            asyncio.get_event_loop().call_soon(cancel_task(run_sidecar))
```

Comment:
why not just await if here? you are already in a separate running coroutine. This way you also avoid overlapping cancel_task requests. Or do you need the speed for the stop requests?

Reply:
I thought that I did not want to cancel the task that actually started the same task. but you may be right these are in the end separate tasks. well I think it does not have a huge impact in the end. I'll let it be.